File: iou-cordapp/src/main/java/com/atfx/sample/schema/IOUContractCriteria.java
package com.atfx.sample.schema;

import net.corda.core.node.services.Vault;
import net.corda.core.node.services.vault.Builder;
import net.corda.core.node.services.vault.QueryCriteria;

public class IOUContractCriteria {
    public QueryCriteria getUserPublicKeyEqualCriteria(String value,
                                                       Vault.StateStatus status) {
        try {
            return new QueryCriteria.VaultCustomQueryCriteria(Builder.equal(
                    PersistentIOU.class.getField("myPublicKey"), value), status);
        } catch (NoSuchFieldException e) {
            //TODO
           //return new QueryCriteria.LinearStateQueryCriteria(null, null, null, Vault.StateStatus.UNCONSUMED, null, Vault.RelevancyStatus.ALL, null);
           return null;
        }
    }
    public QueryCriteria getUserPublicKeyEqualCriteria(String value) {
        return getUserPublicKeyEqualCriteria(value, Vault.StateStatus.UNCONSUMED);
    }
}

File: iou-cordapp/src/main/kotlin/com/formssihk/sample/schema/IOUContractCriteria.kt
package com.formssihk.sample.schema

import net.corda.core.node.services.Vault
import net.corda.core.node.services.vault.QueryCriteria
import net.corda.core.node.services.vault.builder

fun getUserPublicKeyEqualCriteria(value: String, status: Vault.StateStatus = Vault.StateStatus.UNCONSUMED ): QueryCriteria {
    val expr = builder { IOUSchemaV1.PersistentIOU::userPublicKey.equal(value) }
    return QueryCriteria.VaultCustomQueryCriteria(expr, status)
}